metadata-load/src/test/java/uk/ac/ebi/ampt2d/metadata/loader/api/SraAnalysisDocumentAndFileTypeLoaderTest.java

```diff
@@ -0,0 +1,118 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.metadata.loader.api;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import uk.ac.ebi.ena.sra.xml.ANALYSISDocument;
+import uk.ac.ebi.ena.sra.xml.AnalysisFileType;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SraAnalysisDocumentAndFileTypeLoaderTest {
+
+    private static final String ANALYSIS_ACCESSION = "ERZ496533";
+    private static final String ANALYSIS_INVALID_ACCESSION = "ERZ4965336";
+    private static final String ANALYSIS_DOCUMENT_XML = "AnalysisDocument.xml";
+    private static final String ANALYSIS_DOCUMENT_NOT_FOUND = "AnalysisDocumentNotFound.xml";
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    private SraObjectLoaderByAccession<ANALYSISDocument> sraObjectLoaderByAccession;
+
+    private SraObjectLoaderFromAnalysisDocument<AnalysisFileType> sraObjectLoaderFromAnalysisDocument;
+
+    @Before
+    public void restTemplateSetup() throws Exception {
+        when(restTemplate.exchange(SraAnalysisDocumentLoader.getEnaApiUrl(),
+                HttpMethod.GET, null, String.class, ANALYSIS_ACCESSION)).
+                thenReturn(new ResponseEntity(new String(Files.readAllBytes(Paths.get(getClass().getClassLoader()
+                        .getResource(ANALYSIS_DOCUMENT_XML).toURI()))), HttpStatus.OK));
+        when(restTemplate.exchange(SraAnalysisDocumentLoader.getEnaApiUrl(),
+                HttpMethod.GET, null, String.class, ANALYSIS_INVALID_ACCESSION)).
+                thenReturn(new ResponseEntity(new String(Files.readAllBytes(Paths.get(getClass().getClassLoader()
+                        .getResource(ANALYSIS_DOCUMENT_NOT_FOUND).toURI()))), HttpStatus.OK));
+        sraObjectLoaderByAccession = new SraAnalysisDocumentLoader(restTemplate);
+        sraObjectLoaderFromAnalysisDocument = new SraAnalysisFileTypeLoader(sraObjectLoaderByAccession);
+    }
+
+    @Test
+    public void testAnalysisLoad() {
+        Map<String, ANALYSISDocument> listOfAnalysisType =
```

Comment:
The variable name is `list` but the type is `Map`. I assume it is a mapping from accession to analyses, so it could be called `accessionsToAnalyses` maybe?

This applies to other methods below, for instance, `mapOfAnalysisToAnalysisFileTypes` could be `analysisAccessionToFiles`. The `types` suffix makes all sound like we are storing `File.class` or similar.